starboard/linux/shared/soft_mic_platform_service.cc

```diff
@@ -48,51 +43,27 @@ const char kJSONFalse[] = "false";
 
 bool Has(const char* name) {
   // Check if platform has service name.
-  return strcmp(name, "com.google.youtube.tv.SoftMic") == 0;
+  return strcmp(name, kSoftMicPlatformServiceName) == 0;
 }
 
-CobaltExtensionPlatformService Open(void* context,
-                                    const char* name,
-                                    ReceiveMessageCallback receive_callback) {
+PlatformServiceImpl* Open(void* context,
+                          ReceiveMessageCallback receive_callback) {
   SB_DCHECK(context);
 
-  CobaltExtensionPlatformService service;
-
-  if (!Has(name)) {
-    SB_LOG(ERROR) << "Open() service name does not exist: " << name;
-    service = kCobaltExtensionPlatformServiceInvalid;
-  } else {
-    SB_LOG(INFO) << "Open() service created: " << name;
-    service =
-        new CobaltExtensionPlatformServicePrivate({context, receive_callback});
-  }
+  PlatformServiceImpl* impl;
 
-#if SB_IS(EVERGREEN_COMPATIBLE)
-  const bool is_evergreen = elf_loader::EvergreenConfig::GetInstance() != NULL;
-#else
-  const bool is_evergreen = false;
-#endif  // SB_IS(EVERGREEN_COMPATIBLE)
-
-  // The name parameter memory is allocated in h5vcc_platform_service::Open()
-  // with new[] and must be deallocated here.
-  // If we are in an Evergreen build, the name parameter must be deallocated
-  // with free (), since new[] will map to malloc()
-  // in an Evergreen build.
-  if (is_evergreen) {
-    free((void*)name);  // NOLINT
-  } else {
-    delete[] name;
-  }
+  SB_LOG(INFO) << "Open() service created: " << kSoftMicPlatformServiceName;
+  SoftMicPlatformServiceImpl* soft_mic_impl =
+      new SoftMicPlatformServiceImpl({context, receive_callback});
```

Comment:
Our old clang 3.9 builders doesn't know how to create a ctor with the ilitialaliztion list. Please add a ctor to the type.
https://github.com/youtube/cobalt/actions/runs/8651949051/job/23723874730?pr=2896#step:6:15616

Reply:
Done.